src/Modal.js

```diff
@@ -119,30 +188,63 @@ const Modal = React.createClass({
   },
 
   componentDidMount() {
+    const doc = domUtils.ownerDocument(this);
+    const win = domUtils.ownerWindow(this);
+
     this._onDocumentKeyupListener =
-      EventListener.listen(domUtils.ownerDocument(this), 'keyup', this.handleDocumentKeyUp);
+      EventListener.listen(doc, 'keyup', this.handleDocumentKeyUp);
+
+    this._onWindowResizeListener =
+        EventListener.listen(win, 'resize', this.handleWindowResize);
+
+    if (this.props.enforceFocus) {
+      this._onFocusinListener = onFocus(this, this.enforceFocus);
+    }
+
+    let container = getContainer(this);
 
-    let container = (this.props.container && React.findDOMNode(this.props.container)) ||
-          domUtils.ownerDocument(this).body;
     container.className += container.className.length ? ' modal-open' : 'modal-open';
 
-    this.focusModalContent();
+    this._containerIsOverflowing = container.scrollHeight > containerClientHeight(container, this);
+
+    this._originalPadding = container.style.paddingRight;
```

Comment:
the `State` is used anyway here
L227 `this.setState(this._getStyles() //eslint-disable-line react/no-did-mount-set-state`
so why mix private fields with it ?

I'm just kinda an apprentice so I just don't know `cons` and `pros` with this approach.
The link to any article where I can learn about why here is used such approach 
(of course if you have any) would be awesome for me. 
Or you can just send me to google with the right search-phrase-query (I just don't know what is the right one)
:cherries: